test/server/cards/01_SOR/events/Bamboozle.spec.ts

```diff
@@ -0,0 +1,71 @@
+describe('Bamboozle', function () {
+    integration(function (contextRef) {
+        it('Bamboozle should be played by discard a Cunning card, its ability should exhaust a unit and return each upgrades to owner hand', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['bamboozle', 'wampa', 'crafty-smuggler', 'lothal-insurgent'],
+                    groundArena: ['battlefield-marine'],
+                    leader: 'lando-calrissian#with-impeccable-taste'
+                },
+                player2: {
+                    groundArena: [{ card: 'saw-gerrera#extremist', upgrades: ['entrenched', 'shield'] }],
+                    spaceArena: ['green-squadron-awing']
+                }
+            });
+
+            const { context } = contextRef;
+
+            context.player1.clickCard(context.bamboozle);
+            expect(context.player1).toHaveExactPromptButtons(['Cancel', 'Play this event', 'Play Bamboozle by discarding a Cunning card']);
+            // play bamboozle with resource
+            context.player1.clickPrompt('Play this event');
+            expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.greenSquadronAwing, context.sawGerrera]);
```

Comment:
Let's check the exhausted resources here just to make sure the standard play wasn't affected somehow

Reply:
line 35 check it